sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs

```diff
@@ -290,57 +141,104 @@ public virtual bool TryGetFailedEntityFromException(RequestFailedException excep
+            _requestMessages.Enqueue(entity.RowKey);
+            _requestLookup[entity.RowKey] = new BatchItem(requestType, entity, null);
         }
 
         /// <summary>
-        /// Builds an ordered list of <see cref="HttpMessage"/>s containing the batch sub-requests.
+        /// Gets the collection of entities represented in the batch.
         /// </summary>
-        /// <returns></returns>
-        private List<(ITableEntity Entity, HttpMessage HttpMessage)> BuildOrderedBatchRequests()
+        /// <exception cref="NotImplementedException"></exception>
+        public IReadOnlyList<ITableEntity> GetEntities()
         {
-            var orderedList = _requestMessages.ToList();
-            foreach (var item in orderedList)
-            {
-                _changeset.AddContent(new RequestRequestContent(item.HttpMessage.Request));
-            }
-            return orderedList;
+            return _requestMessages
+                .ToList()
+                .Select(rk => _requestLookup[rk].Entity)
+                .ToList()
+                .AsReadOnly();
         }
 
         /// <summary>
-        /// Creates a pipeline to use for processing sub-operations before they
-        /// are combined into a single multipart request.
+        /// Clears the contents of the batch operation.
         /// </summary>
-        /// <returns>A pipeline to use for processing sub-operations.</returns>
-        private static HttpPipeline CreateBatchPipeline()
+        public void ClearOperations()
         {
-            // Configure the options to use minimal policies
-            var options = new TableClientOptions();
-            options.Diagnostics.IsLoggingEnabled = false;
-            options.Diagnostics.IsTelemetryEnabled = false;
-            options.Diagnostics.IsDistributedTracingEnabled = false;
-            options.Retry.MaxRetries = 0;
+            _requestLookup.Clear();
+            _requestMessages = new ConcurrentQueue<string>();
+        }
 
-            // Use an empty transport so requests aren't sent
-            options.Transport = new MemoryTransport();
+        /// <summary>
+        /// Removes the specified entity from the batch.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public bool RemoveEntityOperation(ITableEntity entity)
+            => RemoveEntityOperations(new[] { entity });
 
-            // Use the same authentication mechanism
-            return HttpPipelineBuilder.Build(options);
+        /// <summary>
+        /// Removes the specified entities from the batch.
+        /// </summary>
+        /// <param name="entities"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public bool RemoveEntityOperations(IEnumerable<ITableEntity> entities)
```

Comment:
How would customers use the return value of this method?

Reply:
If, for some reason, we failed to remove the operations they could try again or inspect the current entities in the batch and make a decision about why it failed.

Reply:
What would the reason be if they passed a set of operations that a part of this batch?

Reply:
Perhaps some are not, in fact, a part of the batch.
This is modeled after `List.Remove` and `Dictionary.Remove`

Reply:
But those don't take enumerables.

Reply:
Are you suggesting it return void or that we return something more helpful, such as a list of the entities that could not be removed?